modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -135,4 +146,34 @@ private String unwrapChildKey(String key)
     {
         return key.replaceAll(CHILD_ELEMENT_KEY_PREFIX, "").replaceAll(CHILD_ELEMENT_KEY_SUFFIX, "");
     }
+
+    private void checkParameterGroupExclusivenessForModel(EnrichableModel model, Set<String> resolverKeys) throws ConfigurationException
+    {
+        Optional<ParameterGroupModelProperty> parameterGroupModelProperty = model.getModelProperty(ParameterGroupModelProperty.class);
+        if (parameterGroupModelProperty.isPresent() && parameterGroupModelProperty.get().hasExclusion())
+        {
+            checkParameterGroupExclusiveness(parameterGroupModelProperty.get(), resolverKeys);
+        }
+    }
+
+    private void checkParameterGroupExclusiveness(ParameterGroupModelProperty parameterGroupModelProperty, Set<String> resolverKeys) throws ConfigurationException
+    {
+        for (ParameterGroup group : parameterGroupModelProperty.getGroups())
+        {
+            Set<String> parametersFromGroup = group.getParameters().stream()
+                    .map(p -> p.getName())
+                    .filter(name -> resolverKeys.contains(name))
+                    .collect(toSet());
+
+            if (parametersFromGroup.size() > 1)
+            {
+                throw new ConfigurationException(createStaticMessage(format("Only one of the following parameters [%s] from class '%s' can be present due to its exclusive relation", Arrays.toString(parametersFromGroup.toArray()), group.getType().getName())));
+            }
+
+            if (group.oneRequired() && parametersFromGroup.isEmpty())
+            {
+                throw new ConfigurationException((createStaticMessage(format("Parameter group class '%s' requires that at least one of its parameter should be present but all of them are missing", group.getType().getName()))));
```

Comment:
fe de erratas: where it says operation it should say "whatever type of thing this is"